crates/solidity/testing/perf/npm/src/runners/slang.runner.mts

```diff
@@ -0,0 +1,115 @@
+import { CompilationBuilder, File } from "@nomicfoundation/slang/compilation";
+import { TerminalKind } from "@nomicfoundation/slang/cst";
+import assert from "node:assert";
+import { Runner, SolidityProject, Timing } from "../common.mjs";
+
+function createBuilder(project: SolidityProject): CompilationBuilder {
+  const builder = CompilationBuilder.create({
+    languageVersion: project.compilation.plainVersion(),
+
+    readFile: async (fileId) => {
+      return project.fileContents(fileId);
+    },
+
+    resolveImport: async (sourceFileId, importPath) => {
+      const importLiteral = importPath.node.unparse();
+      assert(importLiteral.startsWith('"') || importLiteral.startsWith("'"));
+      assert(importLiteral.endsWith('"') || importLiteral.endsWith("'"));
+
+      let importString = importLiteral.slice(1, -1);
+      return project.resolveImport(sourceFileId, importString);
+    },
+  });
+
+  return builder;
+}
+
+enum BindingsTarget {
+  File,
+  Project,
+}
+
+class SlangRunner implements Runner {
+  public name = "slang";
+  target: BindingsTarget;
+
+  public constructor(target: BindingsTarget) {
+    this.target = target;
+  }
+
+  async test(project: SolidityProject, file: string): Promise<Timing[]> {
+    const startTime = performance.now();
+    const builder = createBuilder(project);
+
+    await builder.addFile(file);
+
+    const unit = builder.build();
```

Comment:
I'm not sure I understand `BindingsTarget` and `SlangBindingsFileRunner` here. Both would be running all steps/reporting the same `Timing[]` with duplicate keys below, right?
Why not just measure the time it takes to resolve the main file in the same runner?